src/Components/Firestore.js

```diff
@@ -211,35 +208,105 @@ export function generateId() {
   return doc(collection(db, "test")).id;
 }
 
+export async function getReviewReactions(uid, uniqueEntityId) {
+  let reactionsRef = doc(db, "users", uid, "reactions", uniqueEntityId);
+  let reactionSnap = await getDoc(reactionsRef);
+  // Use default value if the document doesn't exist yet.
+  if (!reactionSnap.data()) {
+    return {
+      uniqueEntityId: uniqueEntityId,
+      applause: false,
+      heart: false,
+      trash: false,
+    };
+  } else {
+    return reactionSnap.data();
+  }
+}
+
 // Handle "watchlistData / reactions" on firestore
 
-export async function GetListReactions(docId, setListRxns) {
-  try {
-    let animeRef = doc(db, "watchlistData", docId, "reactions", "emojis");
-    let querySnapshot = await getDoc(animeRef);
+export async function getUserReactions(uid, docId) {
+  let reactionsRef = doc(db, "users", uid, "reactions", docId);
+  let reactionSnap = await getDoc(reactionsRef);
+  if (!reactionSnap.data()) {
     // Use default value if the document doesn't exist yet.
-    if (!querySnapshot.data()) {
-      setListRxns({
-        emojis: { applause: [], heart: [], trash: [] },
-      });
-      return;
-    }
-    let temp = querySnapshot.data();
-    setListRxns(temp);
-  } catch (error) {
-    console.error("Error loading data from Firebase Database", error);
+    return {
+      uniqueEntityId: docId,
+      applause: false,
+      heart: false,
+      trash: false,
+    };
+  } else {
+    return reactionSnap.data();
   }
 }
 
-export async function SaveListReactionsToFirestore(docId, updatedRxns) {
-  let Ref = doc(db, "watchlistData", docId, "reactions", "emojis");
+export async function getReactionCount(uid, docId) {
+  let rxnCountRef = doc(db, "watchlistData", docId);
+  let rxnCountSnap = await getDoc(rxnCountRef);
+  return rxnCountSnap.data();
+}
+
+export async function SaveReactionsToFirestore(
+  uid,
+  docId,
+  updatedRxns,
+  reactionTo,
+  reaction,
+  commentOwnerId
+) {
+  const value = updatedRxns[reaction] ? 1 : -1;
+  let key;
+  if (reaction === "trash") key = "trashCount";
+  else if (reaction === "heart") key = "heartCount";
+  else if (reaction === "applause") key = "applauseCount";
+
+  const uniqueEntityId =
+    reactionTo === "comments" ? docId + commentOwnerId : docId;
+
+  const rxnRef = doc(db, "users", uid, "reactions", uniqueEntityId);
+  let rxnCountRef;
+
+  if (reactionTo === "list") {
+    rxnCountRef = doc(db, "watchlistData", docId);
+  } else if (reactionTo === "reviews") {
+    rxnCountRef = doc(db, "animeData", docId, "reviews", uid);
+  } else if (reactionTo === "comments") {
+    rxnCountRef = doc(db, "watchlistData", docId, "reviews", commentOwnerId);
+  }
```

Comment:
Similarly, should we throw if `reactionTo` is not known, since we would be proceeding with an undefined `rxnCountRef`, which will surely cause problems?